dlib/dnn/cuda_data_ptr.h

```diff
@@ -149,6 +149,32 @@ namespace dlib
 
     // ------------------------------------------------------------------------------------
 
+        class resizable_buffer
```

Comment:
Well, but cuda_data_void_ptr has cuda in the name.  Maybe for API consistency it should be resizable_cuda_buffer.